kayenta/kayenta-integration-tests/src/test/java/com/netflix/kayenta/tests/BaseIntegrationTest.java

```diff
@@ -16,26 +16,125 @@
 package com.netflix.kayenta.tests;
 
 import com.netflix.kayenta.Main;
+import com.netflix.kayenta.configuration.EmbeddedPrometheusBootstrapConfiguration;
 import com.netflix.kayenta.configuration.MetricsReportingConfiguration;
+import com.netflix.kayenta.prometheus.config.PrometheusConfigurationProperties;
+import com.netflix.kayenta.prometheus.config.PrometheusManagedAccount;
+import com.netflix.kayenta.prometheus.config.PrometheusResponseConverter;
+import com.netflix.kayenta.prometheus.service.PrometheusRemoteService;
+import com.netflix.kayenta.retrofit.config.RetrofitClientFactory;
+import com.netflix.kayenta.security.AccountCredentials;
+import com.netflix.kayenta.security.AccountCredentialsRepository;
+import java.io.IOException;
+import java.util.Set;
+import okhttp3.OkHttpClient;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.TestInstance;
 import org.junit.jupiter.api.extension.ExtendWith;
-import org.springframework.beans.factory.annotation.Value;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.test.autoconfigure.actuate.metrics.AutoConfigureMetrics;
 import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.context.annotation.Import;
+import org.springframework.core.env.Environment;
 import org.springframework.test.context.ActiveProfiles;
 import org.springframework.test.context.junit.jupiter.SpringExtension;
 
 @AutoConfigureMetrics
 @SpringBootTest(
     classes = {MetricsReportingConfiguration.class, Main.class},
-    webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT,
+    webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT,
     value = "spring.application.name=kayenta")
 @ExtendWith(SpringExtension.class)
 @ActiveProfiles({"base", "prometheus", "graphite", "cases"})
+@Import(EmbeddedPrometheusBootstrapConfiguration.class)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
```

Comment:
Please add a comment explaining why this is here.

Reply:
https://github.com/spinnaker/spinnaker/pull/7053/commits/935664dec54f435296c39f8dcd350f4110821063

Added a comment